test/jasmine/tests/bar_test.js

```diff
@@ -612,6 +702,168 @@ describe('A bar plot', function() {
 
     afterEach(destroyGraphDiv);
 
+    function getAllTraceNodes(node) {
+        return node.querySelectorAll('g.points');
+    }
+
+    function getAllBarNodes(node) {
+        return node.querySelectorAll('g.point');
+    }
+
+    function assertTextIsInsidePath(textNode, pathNode) {
+        var textBB = textNode.getBoundingClientRect(),
+            pathBB = pathNode.getBoundingClientRect();
+
+        expect(pathBB.left).not.toBeGreaterThan(textBB.left);
+        expect(textBB.right).not.toBeGreaterThan(pathBB.right);
+        expect(pathBB.top).not.toBeGreaterThan(textBB.top);
+        expect(textBB.bottom).not.toBeGreaterThan(pathBB.bottom);
```

Comment:
very nice here.

Asserting `getBoundingClientRect` return values would be very susceptible to cross-browser cross-env discrepancies.